components/server/src/workspace/workspace-starter.ts

```diff
@@ -633,6 +633,23 @@ export class WorkspaceStarter {
             configuration.desktopIdeImage = choose.desktopIdeImage;
         }
 
+        log.info(
+            { context: workspace.context, isEditorContext: WithEditorContext.is(workspace.context) },
+            "check context",
+        );
+        if (WithEditorContext.is(workspace.context)) {
```

Comment:
Should not we rather merge it with if above and only compute `ideChoice` and `useLatest` based on it?